### ISSUE TITLE
Disallow operating on stacks that do not belong to us

### ISSUE THREAD

**Issue by anthonyrisinger** [convox/rack#379]
Recently I used:

```
# convox apps params set --app=WRONGRACK InstanceCount=1
```

To manually scale a broken api build refusing to update itself. Unfortunately, I scaled the wrong rack! It would be nice if convox confirmed the target stack matches one of:

```
stack.Name == os.Getenv("RACK")
stack.Tags["Rack"] == os.Getenv("RACK")
```

to avoid operating on stacks not owned by the current rack.

**Comment by nzoschke** [comment]
I strongly support this. Accidentally doing stuff to other stacks could be disastrous.

**Comment by nzoschke** [comment]
The API that is serving a request knows it's own rack name from os.Getenv("RACK"). So the GetApp or GetRack calls should verify that they only access and operate on stacks that belong to the rack. This can be validated by the stack name itself, as well as the tags that stack has.

**Comment by nzoschke** [comment]
The most important things to validate are anything that will change a stack like a CloudFormation StackUpdate or StackDelete call

**Comment by MiguelMoll** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;7sIdDSju&#x2F;251-disallow-operating-on-stacks-that-do-not-logged-into"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Disallow operating on stacks that do not logged into</a>

**Comment by MiguelMoll** [comment]
Merged with PR #770
